src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImpl.java

```diff
@@ -184,7 +184,27 @@ private HashMap<String, String> changeStatus(String rootOrg, String org, WfReque
 			applicationStatus.setComment(wfRequest.getComment());
 			applicationStatus.setServiceName(serviceName);
 			addModificationEntry(applicationStatus,userId,wfRequest.getAction(),role);
+			String fieldKey = null;
 			wfStatusRepo.save(applicationStatus);
+			List<HashMap<String, Object>> updatedValueList = wfRequest.getUpdateFieldValues();
+			for(Map<String, Object> updatedValue : updatedValueList){
+				if(updatedValue.containsKey(Constants.TO_VALUE)){
+					Map<String, Object> toValue = (Map<String, Object>) updatedValue.get(Constants.TO_VALUE);
+					fieldKey = toValue.entrySet().stream().findFirst().get().getKey();
+				}
+			}
+			if(!StringUtils.isEmpty(fieldKey) && Constants.NAME.equalsIgnoreCase(fieldKey)){
+				Map<String, Object> propertyMap = new HashMap<>();
+				propertyMap.put(Constants.USER_ID, wfRequest.getApplicationId());
+				List<Map<String, Object>> userDetails = cassandraOperation.getRecordsByProperties(
+						Constants.KEYSPACE_SUNBIRD, Constants.USER_TABLE, propertyMap, Arrays.asList(Constants.ROOT_ORG_ID));
+				String rootOrgId = null;
+				if (!CollectionUtils.isEmpty(userDetails)) {
+					rootOrgId = (String) userDetails.get(0).get(Constants.USER_ROOT_ORG_ID);
+				}
+				wfRequest.setRootOrgId(rootOrgId);
+
+			}
```

Comment:
this is going to be a race condition between processing notification request and processing approval request..
if applicationConsumer class executes first - what will happen? user table would be updated first and it would point to the new MDO only.
your test cases worked because you were handling only one request - but that won't be the case in PROD.
you need to enhance the wfRequest object as discussed - from serviceLayer and add the oldRootOrgId (i.e. before pushing kafka message) and pass it on to the consumer class.
just simply check oldRootOrgId exist in wfRequest - if so send out email